src/Compilers/CSharp/Test/Emit2/CodeGen/CodeGenRefReadonlyParameterTests.cs

```diff
@@ -0,0 +1,566 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen;
+
+public class CodeGenRefReadonlyParameterTests : CSharpTestBase
+{
+    private const string RequiresLocationAttributeName = "RequiresLocationAttribute";
+    private const string RequiresLocationAttributeNamespace = "System.Runtime.CompilerServices";
+    private const string RequiresLocationAttributeQualifiedName = $"{RequiresLocationAttributeNamespace}.{RequiresLocationAttributeName}";
+
+    private static void VerifyRequiresLocationAttributeSynthesized(ModuleSymbol module)
+    {
+        var attributeType = module.GlobalNamespace.GetMember<NamedTypeSymbol>(RequiresLocationAttributeQualifiedName);
+        if (module is SourceModuleSymbol)
+        {
+            Assert.Null(attributeType);
+        }
+        else
+        {
+            Assert.NotNull(attributeType);
+        }
+    }
+
+    [Fact]
+    public void Method()
+    {
+        var source = """
+            class C
+            {
+                public void M(ref readonly int p) { }
+            }
+            """;
+        var verifier = CompileAndVerify(source, targetFramework: TargetFramework.NetStandard20,
+            sourceSymbolValidator: verify, symbolValidator: verify);
+        verifier.VerifyDiagnostics();
+        verifier.VerifyMethodIL("C", "M", """
```

Comment:
>verifier.VerifyMethodIL("C", "M", """ [](http://example.com/codeflow?start=8&length=37)

It doesn't look like there is anything interesting to verify in IL. Correct metadata we verify through APIs. We have API to check RefKind, presence of attributes, custom modifiers, etc. We are trying to avoid comparing IL when possible. Among other reasons, the expectations are not obvious and reviewers are not sure on what part of IL to focus. Reviewing big textual baselines is complicated and, often, not done as thorough.